oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -126,7 +136,18 @@ impl ChannelWriter {
         if self.is_orphan() {
             return Err(OakStatus::ERR_CHANNEL_CLOSED);
         }
+
         let mut messages = self.messages.write().unwrap();
+        let mut waiting_threads = self.waiting_threads.lock().unwrap();
```

Comment:
Is the nested locking required? This looks like it imposes a locking hierarchy (`messages` ⊲ `waiting_threads`) that's not documented and so might catch someone out with a deadlock in future.

I'd have guessed that individual blocks would work, but I've not thought too deeply:
```Rust
{
  let mut messages = self.messages.write().unwrap();
  messages.push_back(msg);
}
// Now the channel has a message, wake up all threads waiting on the channel.
// Only one will get the message, but the other threads will cope regardless.
{
  let mut waiting_threads = self.waiting_threads.lock().unwrap();
  for thread in waiting_threads {
     ...
  }
}
Ok(())
```